apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js

```diff
@@ -153,8 +161,33 @@ const AllResidentBillingReceiptsService = new GQLCustomSchema('AllResidentBillin
                     deletedAt: null,
                 })
 
+                // cache billing receipt for calculation of is payable field
+                const minPeriod = min(receiptsForConsumer.map(item => item.period))
+                const nextPeriodBillingReceiptsCache = await find('BillingReceipt', {
+                    account: { id_in: [...new Set(receiptsForConsumer.map(item => item.account.id))], deletedAt: null },
+                    OR: [
+                        { receiver: { AND: [{ id_in: [...new Set(receiptsForConsumer.map(item => item.account.id))] }, { deletedAt: null } ] } },
+                        { category: { AND: [{ id_in: [...new Set(receiptsForConsumer.map(item => item.category.id))] }, { deletedAt: null } ] } },
+                    ],
+                    period_gt: minPeriod,
+                    deletedAt: null,
+                })
+
+                // cache acquiring integration contexts
+                const aicIds = serviceConsumersWithBillingAccount.map(consumer => consumer.acquiringIntegrationContext)
+                    .filter(aic => !isNil(aic))
+                const acquiringContexts = await AcquiringIntegrationContext.getAll(context, {
```

Comment:
AcquiringContext can be wrong, as when organization changes acquiringContext - no changes will be in serviceConsumer